### ISSUE TITLE
Error on 'ember serve' - unable to locate jQuery sourcemap

### ISSUE THREAD

**Issue by mharrisondev** [aupac/ember-aupac-typeahead#55]
Installed via ember cli.
Attempted a clean install of node_modules and bower files, made no difference.

yarn dependencies:
"ember-cli": "2.14.0"
"ember-aupac-typeahead": "git://github.com/aupac/ember-aupac-typeahead#master"

Launching project yields the following error:

``` Javascript
ember serve
Livereload server on http://localhost:49153
Serving on http://localhost:4200/
The Broccoli Plugin: [SourceMapConcat: Concat: Vendor /assets/vendor.js] failed with:
Error: ENOENT: no such file or directory, open '*PROJECT*/tmp/source_map_concat-input_base_path-vS6fJHUI.tmp/vendor/typeahead.jquery.min.js'
    at Object.fs.openSync (fs.js:558:18)
    at Object.fs.readFileSync (fs.js:468:33)
    at SourceMap.addFile (*PROJECT*/node_modules/fast-sourcemap-concat/lib/source-map.js:75:31)
    at *PROJECT*/node_modules/broccoli-concat/concat.js:200:16
    at Array.forEach (native)
    at Concat.<anonymous> (*PROJECT*/node_modules/broccoli-concat/concat.js:198:24)
    at *PROJECT*/node_modules/fast-sourcemap-concat/lib/source-map.js:393:12
    at initializePromise (*PROJECT*/node_modules/rsvp/dist/rsvp.js:589:5)
    at new Promise$1 (*PROJECT*/node_modules/rsvp/dist/rsvp.js:1077:33)
    at SourceMap.end (*PROJECT*/node_modules/fast-sourcemap-concat/lib/source-map.js:380:10)

The broccoli plugin was instantiated at:
    at Concat.Plugin (*PROJECT*/node_modules/broccoli-plugin/index.js:7:31)
    at new Concat (*PROJECT*/node_modules/broccoli-concat/concat.js:38:10)
    at module.exports (*PROJECT*/node_modules/broccoli-concat/index.js:26:10)
    at EmberApp.concatFiles (*PROJECT*/node_modules/ember-cli/lib/broccoli/ember-app.js:397:10)
    at EmberApp._concatFiles (*PROJECT*/node_modules/ember-cli/lib/broccoli/ember-app.js:410:15)
    at EmberApp.javascript (*PROJECT*/node_modules/ember-cli/lib/broccoli/ember-app.js:1286:12)
    at EmberApp.toArray (*PROJECT*/node_modules/ember-cli/lib/broccoli/ember-app.js:1668:10)
    at EmberApp.toTree (*PROJECT*/node_modules/ember-cli/lib/broccoli/ember-app.js:1690:30)
    at module.exports (*PROJECT*/ember-cli-build.js:52:14)
    at Builder.setupBroccoliBuilder (*PROJECT*/node_modules/ember-cli/lib/models/builder.js:54:19)
```

**Comment by stickbyatlas** [comment]
I attempted to install this addon and got the same error above when rebooting my application. @mharrisondev Did you find a workaround?

**Comment by jackmatt2** [comment]
@mharrisondev @stickbyatlas 

It appears that it is having trouble with the bower dependency.

Below is a snippet from `index.js` which is where the files you are missing are included.

I am guessing that `app.bowerDirectory` is either missing, redefined or null for some reason.  Might be worth adding some `console.log` statements and having a look at what is happening.

```javascript
  treeForVendor: function(defaultTree) {
    const app = this._findHost();
    const assetPath = path.join(app.bowerDirectory, 'corejs-typeahead/dist/');

    if (fs.existsSync(assetPath)) {
      let jquerymin = new Funnel(assetPath, {
        files: ['typeahead.jquery.min.js']});
      jquerymin = map(jquerymin,
        (content) => `if (typeof FastBoot === 'undefined') { ${content} }`);
      return new mergeTrees([defaultTree, jquerymin]);
    } else {
      return defaultTree;
    }
```

**Comment by jackmatt2** [comment]
Might also be related to 

> As of Ember 2.12, Ember no longer installs Bower

as mentioned in #50 

Therefore updating the plugin to use NPM instead of BOWER might solve this issue.  I would accept a pull request that makes this change.

**Comment by carrotalan** [comment]
We are also getting this issue.

This would mean as it stands, this addon is incompatible with Ember 2.12 and above

**Comment by jackmatt2** [comment]
Yes, agree.

**Comment by jackmatt2** [comment]
Currently working on this.

**Comment by mharrisondev** [comment]
Hi crew, sorry for the late reply.
We are no longer using this dependency in our project :cold_sweat:
I hope your changes work out :+1:

**Comment by jackmatt2** [comment]
FIXED ember-aupac-typeahead@3.0.0
